pkg/component/worker/utils_linux.go

```diff
@@ -0,0 +1,21 @@
+/*
+Copyright 2023 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package worker
+
+func winExecute(args ...string) error {
```

Comment:
Wouldn't it be sufficient to not define this function at all? Defining it like this makes it a runtime error rather than a compile time error.

Reply:
Ah, now I understand. It's guarded via `if runtime.GOOS` on the call sites instead of using conditional compilation. I'd suggest to either use runtime guards or conditional compilation, but not both. So either capture the call sites via conditional compilation or remove this file and add an `if runtime.GOOS == "windows" { panic(...) }` at the top of the real `winExecute` definition.

Reply:
Why not to use both, if it solves an issue? Plus it makes code easier